src/utils/requestParams.ts

```diff
@@ -0,0 +1,88 @@
+export default class RequestParams<R extends Object> {
+  private criteria: { [index: string]: string } = {};
+
+  limit(limit: number): RequestParams<R> {
+    this.criteria['_limit'] = String(limit);
+    return this;
+  }
+
+  offset(offset: number): RequestParams<R> {
+    this.criteria['_offset'] = String(offset);
+    return this;
+  }
+
+  fields(...values: string[]): RequestParams<R> {
+    this.criteria['_fields'] = values.join(',');
+    return this;
+  }
+
+  eq<F extends keyof R>(field: F, value: string): RequestParams<R> {
```

Comment:
`<F extends keyof R>(field: F` should be added to all of the methods with a field parameter